src/Authentication/Authentication/Helpers/WebResponseHelper.cs

```diff
@@ -34,5 +33,93 @@ internal static class WebResponseHelper
 
             return headers;
         }
+
+        /// <summary>
+        /// Clones an HttpRequestMessage (without the content)
+        /// </summary>
+        /// <param name="original">Original HttpRequestMessage (Will be diposed before returning)</param>
+        /// <param name="requestUri"></param>
+        /// <param name="method"></param>
+        /// <returns>A clone of the HttpRequestMessage</returns>
+        private static HttpRequestMessage CloneHttpRequest(this HttpRequestMessage original, Uri requestUri = null, HttpMethod method = null)
+        {
+            var clone = new HttpRequestMessage
+            {
+                Method = method ?? original.Method,
+                RequestUri = requestUri ?? original.RequestUri,
+                Version = original.Version,
+            };
+
+            foreach (var prop in original.Properties)
+            {
+                clone.Properties.Add(prop);
+            }
+
+            foreach (var header in original.Headers)
+            {
+                /*
+                **temporarily skip cloning telemetry related headers**
+                clone.Headers.TryAddWithoutValidation(header.Key, header.Value);
+                */
+                if (!"x-ms-unique-id".Equals(header.Key) && !"x-ms-client-request-id".Equals(header.Key) && !"CommandName".Equals(header.Key) && !"FullCommandName".Equals(header.Key) && !"ParameterSetName".Equals(header.Key) && !"User-Agent".Equals(header.Key))
```

Comment:
Do we really use these request headers in our clients? Here is the complete list of all the headers we send https://github.com/microsoftgraph/msgraph-sdk-design/blob/master/middleware/TelemetryHandler.md#ideal-metadata-capture.